astroquery/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py

```diff
@@ -68,44 +72,63 @@
 
 
 def mock_get(self, method, url, *args, **kwargs):  # pragma: nocover
-    assert url == conf.url
 
+    # Read request parameters
     params = kwargs.get("params", None)
     assert params is not None
 
     try:
+        # Read request (URL substring for API or SQL query for TAP) from json file as dict
         with open(RESPONSE_FILE, "r") as f:
             responses = json.load(f)
     except FileNotFoundError:
         responses = {}
 
-    # Work out where the expected response is saved
     table = params["table"]
-    key = urlencode(sorted(params.items()))
+    # Set base URL depending on whether TAP or API for access
```

Comment:
re: tests for TAP interface. From the experiments I've tried, I think it would be worth reverting these changes to `mock_get` and then write separate test cases for the TAP API that follow the examples here: https://github.com/astropy/astroquery/blob/main/astroquery/cadc/tests/test_cadctap.py

It seems like it would be possible to update this function to do what we need, but it might be more trouble than it's worth.

Reply:
Yeah it started getting messy, so that is probably a good idea. Thanks!

Reply:
Done!